playground/backend/internal/executors/java_helper.go

```diff
@@ -0,0 +1,186 @@
+func NewJavaExecutor(fs *fs_tool.LifeCycle, javaValidators *[]validatorWithArgs, javaPreparation *[]preparationWithArgs) *Executor {
+	compileArgs := []string{"-d", binFolder, "-classpath", beamJarPath}
+	fullClassPath := strings.Join([]string{binFolder, beamJarPath, runnerJarPath, slf4jPath}, ":")
+	runArgs := []string{"-cp", fullClassPath}
+	if javaValidators == nil {
+		v := make([]validatorWithArgs, 0)
+		javaValidators = &v
+	}
+	path, _ := os.Getwd()
+
+	exec := new(Executor)
+	exec.validators = *javaValidators
+	exec.preparation = *javaPreparation
+	exec.relativeFilePath = fs.GetRelativeExecutableFilePath()
+	exec.absoulteFilePath = fs.GetAbsoluteExecutableFilePath()
+	exec.dirPath = filepath.Join(path, fs.Folder.BaseFolder)
+	exec.compileName = javaCompileCmd
+	exec.runName = javaRunCmd
+	exec.compileArgs = compileArgs
+	exec.runArgs = runArgs
+	return exec
+}
+
+// GetJavaValidators return validators methods that needed for Java file
+func GetJavaValidators() *[]validatorWithArgs {
+	validatorArgs := make([]interface{}, 1)
+	validatorArgs[0] = javaExtension
+	pathCheckerValidator := validatorWithArgs{
+		validator: fs_tool.CheckPathIsValid,
+		args:      validatorArgs,
+	}
+	validators := []validatorWithArgs{pathCheckerValidator}
+	return &validators
+}
+
+// GetJavaPreparation return validation methods that needed for Java file
+func GetJavaPreparation() *[]preparationWithArgs {
+	publicClassModification := preparationWithArgs{
+		prepare: removePublicClassModification,
+	}
+	additionalPackage := preparationWithArgs{
+		prepare: removeAdditionalPackage,
+	}
+	validators := []preparationWithArgs{publicClassModification, additionalPackage}
+	return &validators
+}
+
+func removeAdditionalPackage(filePath string, args ...interface{}) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		grpclog.Errorf("Preparation:  remove additional package: Error during open file: %s", filePath)
+		return err
+	}
+
+	tmp, err := os.Create("temporary.java")
+	if err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during create new temporary file")
+		return err
+	}
+
+	if err := removePackageString(file, tmp); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during move data from original file to to temporary file")
+		return err
+	}
+
+	if err := tmp.Close(); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during Close temporary file")
+		return err
+	}
+	if err := file.Close(); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during Close original file")
+		return err
+	}
+
+	pathSlice := strings.Split(filePath, "/")
+	fileName := pathSlice[len(pathSlice)-1]
+	if err := os.Rename(tmp.Name(), fileName); err != nil {
+		grpclog.Error("Preparation: remove additional package: Error during rename temporary file")
+		return err
+	}
+	return nil
+}
+
+func removePackageString(reader io.Reader, writer io.Writer) error {
```

Comment:
Might the package not be a string? What is the reason to name it String postfix?